compiler/src/iree/compiler/Dialect/LinalgExt/Transforms/SplitReduction.cpp

```diff
@@ -389,4 +390,332 @@ splitReduction(RewriterBase &rewriter, LinalgExt::TopkOp topkOp,
   return success();
 }
 
+struct ArgmaxCombinerOps {
+  Operation *maxOp = nullptr;    // arith.maximumf
+  Operation *selectOp = nullptr; // arith.select
+  Operation *cmpOp = nullptr;    // arith.cmpf
+};
+
+// Matches the combiner pattern in a linalg.generic argmax-style reduction:
+// Example MLIR:
+// %4:2 = linalg.generic {
+//     indexing_maps = [...],
+//     iterator_types = ["parallel", "reduction"]
+//   } ins(%arg0 : tensor<?x128xbf16>) outs(%1, %3 : tensor<?xbf16>,
+//   tensor<?xi64>) {
+// ^bb0(%in: bf16, %out: bf16, %out_0: i64):
+//   %5 = linalg.index 1 : index
+//   %6 = arith.index_cast %5 : index to i64
+//   %7 = arith.maximumf %in, %out : bf16
+//   %8 = arith.cmpf ogt, %in, %out : bf16
+//   %9 = arith.select %8, %6, %out_0 : i64
+//   linalg.yield %7, %9 : bf16, i64
+// } -> (tensor<?xbf16>, tensor<?xi64>)
+//
+// This function extracts the `arith.maximumf`, `arith.cmpf`, and `arith.select`
+// operations from the body to facilitate transformations such as split
+// reduction.
+static ArgmaxCombinerOps collectArgmaxCombinerOps(linalg::GenericOp genericOp) {
+  assert(isArgmaxOp(genericOp) && "expected operation to be an argmax op");
+
+  auto yieldOp = cast<linalg::YieldOp>(genericOp.getBody()->getTerminator());
+
+  // Extract max value producer: arith.maximumf.
+  Value maxResult = yieldOp.getOperand(0);
+  auto maxOp = cast<arith::MaximumFOp>(maxResult.getDefiningOp());
+
+  // Extract index result producer: arith.select.
+  Value indexResult = yieldOp.getOperand(1);
+  auto selectOp = cast<arith::SelectOp>(indexResult.getDefiningOp());
+
+  // Extract the condition of the select, expected to be arith.cmpf with
+  // predicate OGT.
+  auto cmpOp = cast<arith::CmpFOp>(selectOp.getCondition().getDefiningOp());
+
+  ArgmaxCombinerOps ops;
+  ops.maxOp = maxOp;
+  ops.selectOp = selectOp;
+  ops.cmpOp = cmpOp;
+  return ops;
+}
+
+static Value expandValue(OpBuilder &builder, Location loc, Value value,
+                         RankedTensorType expandedType) {
+  RankedTensorType originalType = cast<RankedTensorType>(value.getType());
+  if (originalType == expandedType)
+    return value;
+
+  auto reassociation =
+      mlir::getReassociationIndicesForReshape(originalType, expandedType);
```

Comment:
Sorry, I missed this in the previous reviews. I think we usually drop `mlir::` prefix in transformation implementation. This is why we have `iree_compiler` under `mlir` namespace.